### PR TITLE
fixed musl conflict

### LINKED ISSUE #75: rdkafka build fails with musl target

### PR DESCRIPTION
**Pull request by pranav-bhatt**
fixes #75

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -59,19 +58,19 @@ jobs:
...
 
-    # If musl, compile and test all excluding kafka
+    # If musl, compile and test all
     - uses: actions-rs/cargo@v1
       if: matrix.target == 'x86_64-unknown-linux-musl'
       with:
-        command: build
         toolchain: ${{ matrix.toolchain }}
-        args: --target ${{ matrix.target }} --workspace --exclude cloudevents-sdk-rdkafka
+    - run: CC=musl-gcc CXX=g++ cargo build --target ${{ matrix.target }} --all
```
Comment (slinkydeveloper) [inline .github/workflows/master.yml:66]: Uhm do we really need `CC` and `CXX` env vars?

Comment (slinkydeveloper) [inline .github/workflows/master.yml:66]: Can you use the cargo action more than manually running the command?

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -59,19 +58,19 @@ jobs:
...
+      if: matrix.target == 'x86_64-unknown-linux-gnu'
     - uses: actions-rs/cargo@v1
       if: matrix.target == 'x86_64-unknown-linux-musl'
       with:
-        command: test
         toolchain: ${{ matrix.toolchain }}
-        args: --target ${{ matrix.target }} --workspace --exclude cloudevents-sdk-rdkafka
+    - run: CC=musl-gcc CXX=g++ cargo test --target ${{ matrix.target }} --all
```
Comment (slinkydeveloper) [inline .github/workflows/master.yml:72]: Why? It's tested above

**Review by slinkydeveloper — COMMENTED** [review]
I think you should review the actions file, it's a little bit messed up

File: .github/workflows/master.yml (hunk 1 above)
Comment (pranav-bhatt) [reply .github/workflows/master.yml:66]: yeah, it didnt work without setting those :(

Comment (pranav-bhatt) [reply .github/workflows/master.yml:66]: the action doesnt allow setting environment vars (do tell me if you know a way)

File: .github/workflows/master.yml (hunk 2 above)
Comment (pranav-bhatt) [reply .github/workflows/master.yml:72]: that was a build run. If you want we could directly test

File: .github/workflows/master.yml (hunk 1 above)
Comment (slinkydeveloper) [reply .github/workflows/master.yml:66]: Using env like this https://docs.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#about-environment-variables doesn't work?

File: .github/workflows/master.yml (hunk 2 above)
Comment (slinkydeveloper) [reply .github/workflows/master.yml:72]: this is a build run using glibc target, we don't need it it's already above

File: .github/workflows/master.yml (hunk 1 above)
Comment (pranav-bhatt) [reply .github/workflows/master.yml:66]: the thing is that these values get passed into the cargo build command or something:
https://github.com/fede1024/rust-rdkafka/issues/88#issuecomment-494130230
(consider this maybe xD: https://github.com/fede1024/rust-rdkafka/issues/88#issuecomment-493685113)

Comment (slinkydeveloper) [reply .github/workflows/master.yml:66]: well, these are just env variables, so if github actions allows you to set them, i don't see why it should not work

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -21,8 +21,7 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - run: sudo apt-get update
-      if: matrix.target == 'x86_64-unknown-linux-musl'
```
Comment (slinkydeveloper) [inline .github/workflows/master.yml:24]: You should not remove this condition: only in musl case we need to update apt packages

File: .github/workflows/master.yml [hunk 4]
```diff
@@ -59,20 +58,25 @@ jobs:
...
 
-    # If musl, compile and test all excluding kafka
+    # If musl, compile and test all
     - uses: actions-rs/cargo@v1
       if: matrix.target == 'x86_64-unknown-linux-musl'
       with:
         command: build
         toolchain: ${{ matrix.toolchain }}
-        args: --target ${{ matrix.target }} --workspace --exclude cloudevents-sdk-rdkafka
+        args: --target ${{ matrix.target }} --all
```
Comment (slinkydeveloper) [inline .github/workflows/master.yml:67]: s/--all/--workspace (--all is being deprecated)

File: .github/workflows/master.yml [hunk 5]
```diff
@@ -59,20 +58,25 @@ jobs:
...
+        CC: musl-gcc
+        CXX: g++
     - uses: actions-rs/cargo@v1
       if: matrix.target == 'x86_64-unknown-linux-musl'
       with:
         command: test
         toolchain: ${{ matrix.toolchain }}
-        args: --target ${{ matrix.target }} --workspace --exclude cloudevents-sdk-rdkafka
-
+        args: --target ${{ matrix.target }} --all
```
Comment (slinkydeveloper) [inline .github/workflows/master.yml:76]: same as above

**Review by slinkydeveloper — COMMENTED** [review]
Small nits, but looks good. I wonder if we should just merge it and check if it works

**Review by slinkydeveloper — APPROVED** [review]

**Pull request merged**